rtos/TARGET_CORTEX/TOOLCHAIN_GCC_ARM/mbed_boot_gcc_arm.c

```diff
@@ -0,0 +1,115 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018-2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cmsis.h"
+#include "mbed_toolchain.h"
+#include "mbed_boot.h"
+#include "mbed_rtos_storage.h"
+#include "cmsis_os2.h"
+
+static osMutexId_t               malloc_mutex_id;
+static mbed_rtos_storage_mutex_t malloc_mutex_obj;
+static osMutexAttr_t             malloc_mutex_attr;
+
+static osMutexId_t               env_mutex_id;
+static mbed_rtos_storage_mutex_t env_mutex_obj;
+static osMutexAttr_t             env_mutex_attr;
+
+#if !defined(HEAP_START)
+/* Defined by linker script */
+extern uint32_t __end__[];
+#define HEAP_START      ((unsigned char*)__end__)
+#define HEAP_SIZE       ((uint32_t)((uint32_t)INITIAL_SP - (uint32_t)HEAP_START))
```

Comment:
Why `INITIAL_SP` is used and not `ISR_STACK_START`.

I remember that `INITIAL_SP` is required for backward compatibility with mbed-rtos, but it would be good to not use it if possible, since `ISR_STACK_START` should define the same address.